src/tplr/chain_sync.py

```diff
@@ -0,0 +1,239 @@
+# The MIT License (MIT)
+# © 2025 tplr.ai
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software,
+# and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+# the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+# THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+# type: ignore
+
+import asyncio
+from typing import Dict, Optional, Tuple
+from collections import defaultdict
+
+import bittensor as bt
+
+from tplr import logger
+from tplr.schemas import Bucket
+
+
+class ChainSync:
+    """Handles blockchain synchronization and peer tracking"""
+
+    def __init__(
+        self,
+        config,
+        netuid=None,
+        metagraph=None,
+        hparams=None,
+        fetch_interval=600,
+        wallet=None,
+    ):
+        self.config = config
+        self.netuid = netuid
+        self.metagraph = metagraph
```

Comment:
I think these should have defaults as well as I suppose we don't want them to be `None` when we are using the methods later?